### ISSUE TITLE
Eslint is not running in CI

### ISSUE THREAD

**Issue by sebastianhaas** [opensheetmusicdisplay/opensheetmusicdisplay#297]
See #274 

Add `npm run lint` to appveyor and Travis builds

**Comment by bneumann** [comment]
I just noticeced that it runs during npm install which is bogus?!

**Comment by sschmidTU** [comment]
Yes, eslint doesn't need to run during npm install, it runs for every commit anyways.

**Comment by bneumann** [comment]
And fun thing, even after searching through hidden folders I can't find any post install scripts that I could modify. @sebastian what sorcery is this?

**Comment by sebastianhaas** [comment]
> I just noticeced that it runs during npm install which is bogus?!

In general, it does run before making a commit, as a result of the pre-commit hook in `package.json`.
After `npm i`, the [`prepare`](https://docs.npmjs.com/misc/scripts#prepublish-and-prepare) hook kicks in and triggers the build and lint. Don't know if we really need that, but we _might_ need it for publish. Haven't really thought that through.

For CI, we should add `npm i` to https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/develop/.travis.yml and https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/develop/.appveyor.yml.

**Comment by sschmidTU** [comment]
That would of course make the builds take a lot longer, but seems reasonable.

**Comment by sebastianhaas** [comment]
Don't think that's a problem. We can discuss the `prepare` though. Maybe we can use something like `prepublish` instead.
